### ISSUE TITLE
create_bucket should fail with CreateBucketConfiguration={}

### ISSUE THREAD

**Issue by eldarbogdanov** [spulec/moto#2210]
The following code produces an obscure exception with real (unmocked) boto3:
```
import boto3
s3_client = boto3.client("s3", region_name="us-east-1")
s3_client.create_bucket(ACL="private", Bucket="eldar-temp", CreateBucketConfiguration={})
```
Result:
```
---------------------------------------------------------------------------
ClientError                               Traceback (most recent call last)
<ipython-input-5-3f69eeffcfc7> in <module>()
----> 1 s3_client.create_bucket(ACL="private", Bucket="eldar-temp", CreateBucketConfiguration={})

/usr/local/lib/python2.7/dist-packages/botocore/client.pyc in _api_call(self, *args, **kwargs)
    312                     "%s() only accepts keyword arguments." % py_operation_name)
    313             # The "self" in this scope is referring to the BaseClient.
--> 314             return self._make_api_call(operation_name, kwargs)
    315
    316         _api_call.__name__ = str(py_operation_name)

/usr/local/lib/python2.7/dist-packages/botocore/client.pyc in _make_api_call(self, operation_name, api_params)
    610             error_code = parsed_response.get("Error", {}).get("Code")
    611             error_class = self.exceptions.from_code(error_code)
--> 612             raise error_class(parsed_response, operation_name)
    613         else:
    614             return parsed_response

ClientError: An error occurred (MalformedXML) when calling the CreateBucket operation: The XML you provided was not well-formed or did not validate against our published schema
```

What boto3 (or rather, underlying s3 API) is unhappy about is the `CreateBucketConfiguration={}` statement.

When this code is mocked with moto's mock_s3, the call to create_bucket succeeds, so its behavior is inconsistent with actual boto3.

I'm using moto==1.3.8.

**Comment by spulec** [comment]
Thanks; setting as a feature request for us to do validation of the bucket config.

**Comment by dawngerpony** [comment]
Since this issue was raised a while ago, it might be worth mentioning that this is still an issue:

```
Traceback (most recent call last):
  File "issue_2210.py", line 3, in <module>
    s3_client.create_bucket(ACL="private", Bucket="eldar-temp", CreateBucketConfiguration={})
  File "/Users/dawn.james/dev/github.com/dawngerpony/moto/.venv/lib/python3.7/site-packages/botocore-1.17.6-py3.7.egg/botocore/client.py", line 316, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/Users/dawn.james/dev/github.com/dawngerpony/moto/.venv/lib/python3.7/site-packages/botocore-1.17.6-py3.7.egg/botocore/client.py", line 635, in _make_api_call
    raise error_class(parsed_response, operation_name)
botocore.exceptions.ClientError: An error occurred (MalformedXML) when calling the CreateBucket operation: The XML you provided was not well-formed or did not validate against our published schema
```

**Comment by dawngerpony** [comment]
The contents of the `ClientError`:

```
{'Error': {'Code': 'MalformedXML', 'Message': 'The XML you provided was not well-formed or did not validate against our published schema'}, 'ResponseMetadata': {'RequestId': '<request ID>', 'HostId': '<HostId>, 'HTTPStatusCode': 400, 'HTTPHeaders': {'x-amz-request-id': '<request ID>', 'x-amz-id-2': '<ID>', 'content-type': 'application/xml', 'transfer-encoding': 'chunked', 'date': '<date>', 'connection': 'close', 'server': 'AmazonS3'}, 'RetryAttempts': 0}}
```

**Comment by dawngerpony** [comment]
As you can see, I've raised #3079 which should fix this issue.

**Comment by bblommers** [comment]
This is now available as of moto >= 1.3.15.dev914
